File: 프로그래머스/lv2/42587. 프로세스/프로세스.java
import java.util.*;

class Position {
    int priority;
    int idx;

    Position(int priority, int idx) {
        this.priority = priority;
        this.idx = idx;
    }
}

class Solution {
    public int solution(int[] priorities, int location) {
        
        int answer = 0;
        Queue<Position> q = new LinkedList<>();
        int[] temp = Arrays.copyOf(priorities, priorities.length);
        Integer[] priArr = Arrays.stream(temp).boxed().toArray(Integer[]::new);
        Arrays.sort(priArr, Collections.reverseOrder());
        
        for (int i = 0; i < priorities.length; i++) {
            q.offer(new Position(priorities[i], i));
        }
        
        int idx = 0;
        while (!q.isEmpty()) {
            Position pos = q.poll();
            
            if (pos.priority == priArr[idx]) {
                answer++;
                idx++;
                
                if (pos.idx == location) {
                    break;
                }
            }
            else {
                q.offer(pos);
            }
        }
        return answer;

    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    answer = 0
    
    queue = deque()
    
    for idx, pri in enumerate(priorities):
        queue.append((pri, idx))
        
    count = 0
    while queue:
        cur_pri, cur_idx = queue.popleft()
        
        if any(cur_pri < other_pri for other_pri, _ in queue):
                queue.append((cur_pri, cur_idx))
                
        else:
                count += 1
                if location == cur_idx:
                    return count
        
    return count